internal/app/synchronizer/evaluator_client.go

```diff
@@ -103,56 +106,43 @@ func (ec *evaluatorClient) initialize() error {
-	resp, err := ec.grpcClient.Evaluate(ctx, &pb.EvaluateRequest{Matches: proposals})
-	if err != nil {
-		return nil, err
-	}
-
-	return resp.GetMatches(), nil
-}
-
-func (ec *evaluatorClient) httpEvaluate(ctx context.Context, proposals []*pb.Match) (results []*pb.Match, err error) {
-	proposalIDs := getMatchIds(proposals)
-	jsonProposals, err := json.Marshal(proposals)
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition, "failed to marshal proposals to string for proposals %s: %s", proposalIDs, err.Error())
-	}
-
-	reqBody, err := json.Marshal(map[string]json.RawMessage{"match": jsonProposals})
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition, "failed to marshal request body for proposals %s: %s", proposalIDs, err.Error())
-	}
-
-	req, err := http.NewRequest("POST", ec.baseURL+"/v1/matches:evaluate", bytes.NewBuffer(reqBody))
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition, "failed to create evaluator http request for proposals %s: %s", proposalIDs, err.Error())
-	}
-
-	resp, err := ec.httpClient.Do(req.WithContext(ctx))
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to get response from evaluator for proposals %s: %s", proposalIDs, err.Error())
-	}
-
-	defer func() {
-		err = resp.Body.Close()
-		if err != nil {
-			evaluatorClientLogger.WithError(err).Warning("failed to close response body read closer")
+func (ec *evaluatorClient) grpcEvaluate(ctx context.Context, proposals []*pb.Match) ([]*pb.Match, error) {
+	stream, err := ec.grpcClient.Evaluate(ctx)
+
+	waitc := make(chan evaluatorResult)
+	defer close(waitc)
+	go func() {
+		results := []*pb.Match{}
+		for {
+			resp, recvErr := stream.Recv()
+			if recvErr == io.EOF {
+				// read done.
+				waitc <- evaluatorResult{results: results, err: nil}
+				return
+			}
+			if recvErr != nil {
+				waitc <- evaluatorResult{results: nil, err: recvErr}
+				return
+			}
+			results = append(results, resp.GetMatch())
 		}
 	}()
 
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition,
-			"failed to read from response body for proposals %s: %s", proposalIDs, err.Error())
+	for _, proposal := range proposals {
+		if err = stream.Send(&pb.EvaluateRequest{Match: proposal}); err != nil {
+			return nil, err
+		}
 	}
 
-	pbResp := &pb.EvaluateResponse{}
-	err = json.Unmarshal(body, pbResp)
-	if err != nil {
-		return nil, status.Errorf(codes.FailedPrecondition,
-			"failed to unmarshal response body to response pb for proposals %s: %s", proposalIDs, err.Error())
+	if err = stream.CloseSend(); err != nil {
+		logger.Errorf("failed to close the send stream: %s", err.Error())
 	}
 
-	return pbResp.GetMatches(), nil
+	result := <-waitc
+	return result.results, result.err
+}
+
+func (ec *evaluatorClient) httpEvaluate(ctx context.Context, proposals []*pb.Match) (results []*pb.Match, err error) {
+	// TODO: Implement a bidirectional HTTP streaming client in v0.7
```

Comment:
Is not implementing this just timing and prioritization? or any technical challenges here?

Reply:
I have #754 depends on this PR that added the HTTP support.